### ISSUE TITLE
[BUG] Incorrect data when saving custom configuration file to database

### ISSUE THREAD

**Issue by weizhuangzhi** [bunkerity/bunkerweb#652]
### What happened?

When I continued to test the issue in [https://github.com/bunkerity/bunkerweb/issues/638](https://github.com/bunkerity/bunkerweb/issues/638), I discovered a strange phenomenon.

After saving the custom configuration file, when I looked at it again, I found that the directory where the file was located was incorrect.

I guess this phenomenon may also be related to the disappearance of files after the service is modified.

### How to reproduce?

Create a new service
![R5` @(Q%N`{OVB@{DBN%3EC](https://github.com/bunkerity/bunkerweb/assets/4264162/bc53d44e-c16a-486e-88ce-b736e9d76ec8)

Create a custom configuration file
![M` R1F}5L`$UM)V$(7~)TP1](https://github.com/bunkerity/bunkerweb/assets/4264162/d46a7fa4-84a4-4d4a-9b7a-5c2007dc1a25)

Storage in database after saving
![AT6LX53OIKQDU0E$D 72EB1](https://github.com/bunkerity/bunkerweb/assets/4264162/8c3caced-7a47-4010-914d-8fff2b342fd0)

Create another custom configuration file
![6L1}0CKTHS_UL$%_F1{`0(O](https://github.com/bunkerity/bunkerweb/assets/4264162/e9e15f34-6bd9-4247-9f34-5a5f9f9cf732)

View another newly added file(the file location is incorrect)
![`YEK PQ L0@2~HG6D 6NBQS](https://github.com/bunkerity/bunkerweb/assets/4264162/3c584c6f-e9af-47bf-a514-c1b5eeeb7627)

At this time, the storage in the database
![XUZOL)}5GCM9$ M{965(@K4](https://github.com/bunkerity/bunkerweb/assets/4264162/d3b5a38c-1b8f-4f38-9751-c1ab6ff1f2cc)

Open the service and save it without modification
![XJ38M5J%(0 _3 IVR}1~JFI](https://github.com/bunkerity/bunkerweb/assets/4264162/35fd864e-e317-42c9-b094-9a9334ed3ced)

All newly added files are gone
![I)KE`E3@FN5CPF873_23RP](https://github.com/bunkerity/bunkerweb/assets/4264162/7fb752c6-3e44-449d-851c-e84f874a7f0f)

At this time, the storage in the database
![U7A_NE@NJPBQ60)7F45U{}C](https://github.com/bunkerity/bunkerweb/assets/4264162/c306ffe5-1590-45e0-9a7f-ed2febaf6021)

### Configuration file(s) (yaml or .env)

```YAML
none
```

### Relevant log output

```shell
none
```

### BunkerWeb version

1.5.2

### What integration are you using?

Linux

### Linux distribution (if applicable)

Fedora 38 (updated to the latest)

### Removed private data

- [X] I have removed all private data from the configuration file and the logs

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by weizhuangzhi** [comment]
@TheophileDiot  using the dev tag, previously created configuration files are no longer deleted when modifying a service.

But I discovered a new problem. When you delete the service and then enter the configs page, a 500 error will appear. The log is:

```
[2023-10-07 11:07:55 +0800] [849565] [❌] Error handling request /manage/configs
Traceback (most recent call last):
  File "/usr/share/bunkerweb/deps/python/gunicorn/workers/gthread.py", line 282, in handle
    keepalive = self.handle_request(req, conn)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/gunicorn/workers/gthread.py", line 334, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/flask/app.py", line 2213, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/ui/src/ReverseProxied.py", line 62, in __call__
    return self.app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/flask/app.py", line 2193, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/ui/main.py", line 716, in configs
    path_to_dict(
  File "/usr/share/bunkerweb/ui/utils.py", line 81, in path_to_dict
    d["children"][config_types.index(type_lower)]["children"][[x["name"] for x in d["children"][config_types.index(type_lower)]["children"]].index(conf["service_id"])]["children"].append(file_info)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: 'mysite.net' is not in list
```

Not sure if it's a new problem or if I updated it incorrectly.

**Comment by weizhuangzhi** [comment]
After deleting the service, the service ID of mysite.net no longer exists in the bw_services table.
![image](https://github.com/bunkerity/bunkerweb/assets/4264162/a05495a1-c78d-4f10-9635-c36a25ffba29)

However, there is still a record with service_id mysite.net in the bw_custom_configs table.
![image](https://github.com/bunkerity/bunkerweb/assets/4264162/d995e2bc-0261-4ec5-80fb-b0b95e0d5a31)

This may be the cause of the 500 error on the configs page.

**Comment by weizhuangzhi** [comment]
@TheophileDiot Thanks, it's been verified that the problem no longer exists, that's great.

**Comment by TheophileDiot** [comment]
Let's keep this issue open until it is merged on master 😁

**Comment by fl0ppy-d1sk** [comment]
Hello @weizhuangzhi,

It should be fixed in the last 1.5.3 release. Don't hesitate to open a new issue if needed.
